Yuliia/most-probable-letters.js

```diff
@@ -0,0 +1,103 @@
+const fs = require('fs');
+const readline = require('readline');
+
+if (process.argv.length < 4) {
+  console.error('Example: node statistics-counter.js filename.txt letter');
+  process.exit(1);
+}
+
+const filePath = process.argv[2];
+const letter = process.argv[3];
+
+if (letter.length !== 1 || !letter.match(/[a-z]/i)) {
+  console.error('The letter argument should be a single alphabet character.');
+  process.exit(1);
+}
+
+const dictionary = new Map();
+
+function createDictionary (line, dictionary) {
+  line = line.toLowerCase().normalize('NFKC').replace(/\n/g, ' ').replace(/\r/g, '').replace(/\s+/g, ' ').replace(/[^a-z ]/g, '');
+  for (let i = 0; i < line.length - 1; i++) {
+    let currentChar = line[i];
+    let nextChar = line[i + 1];
+
+    if (!dictionary.has(currentChar)) {
+      dictionary.set(currentChar, new Map([[nextChar, 1]]));
+    } else {
+      let nextCharMap = dictionary.get(currentChar);
+      if(!nextCharMap.has(nextChar)) {
+        nextCharMap.set(nextChar, 1);
+      } else {
+        nextCharMap.set(nextChar, nextCharMap.get(nextChar) + 1);
+      }
+    }
+  }
+  return dictionary;
+}
+
+function findMostProbableLetters (letter, dictionary, iterations = 100) {
```

Comment:
It's code for Step5?)) why we have hardcoded 100 iterations here?)

Reply:
No, step 3)